app/src/main/java/com/github/kpdandroid/ui/screens/FileAnalysisScreen.kt

```diff
@@ -0,0 +1,231 @@
+import com.github.kpdandroid.ui.DetectionResultColumn
+import com.github.kpdandroid.ui.DualBottomMenu
+import com.github.kpdandroid.ui.ExpandableBottomMenuItem
+import com.github.kpdandroid.ui.ExpandableBottomMenuItemContent
+import com.github.kpdandroid.ui.viewmodels.FileAnalysisViewModel
+import com.github.kpdandroid.utils.detection.DetectionAlgo
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.launch
+
+@Composable
+// - Not using `vm = viewModel()` as we need to use exactly the same instance as the MainActivity
+// does.
+// - Not using `vm = viewModel(LocalContext.current as ViewModelStoreOwner)` as it may cause
+// unexpected cast exceptions.
+fun FileAnalysisScreen(vm: FileAnalysisViewModel) {
+    var openDialog by rememberSaveable { mutableStateOf(false) }
+    RunConfigurationDialog(
+        open = openDialog,
+        onDismiss = { openDialog = false },
+        onConfirm = { enteredNum ->
+            vm.startDetection(enteredNum)
+            openDialog = false
+        }
+    )
+
+    Scaffold(
+        bottomBar = {
+            val context = LocalContext.current
+            FileAnalysisMenu(
+                algorithmsItem = ExpandableBottomMenuItemContent(
+                    options = DetectionAlgo.titles,
+                    selectedOption = vm.prefs.fileAlgoTitle,
+                    onSelected = { algoTitle -> vm.prefs.fileAlgoTitle = algoTitle }
+                ),
+                onImagePicked = { vm.setupImage(context.contentResolver.openInputStream(it)) },
+                onLogPicked = { vm.setupLogger(context.contentResolver.openOutputStream(it)) }
+            )
+        },
+        floatingActionButton = {
+            val scope = rememberCoroutineScope { Dispatchers.Default }
+            FileAnalysisFab(
+                show = vm.imageLayers != null && vm.prefs.fileAlgoTitle != DetectionAlgo.NONE.title,
+                progress = vm.detectionProgress,
+                onStart = { openDialog = true },
+                onStop = { scope.launch { vm.stopDetection() } }
+            )
+        },
+        floatingActionButtonPosition = FabPosition.Center,
+        isFloatingActionButtonDocked = true
+    ) { paddingValues ->
+        DetectionResultColumn(
+            imageLayers = vm.imageLayers?.toList() ?: emptyList(),
+            altText = "Pick an image for analysis",
+            captions = vm.calcTimesMs?.let { (calcTimeMs, stdDevMs) ->
+                listOf(
+                    "Mean detection time: ${formatTime(calcTimeMs)} ms.",
+                    "Standard deviation: ${formatTime(stdDevMs)} ms."
+                )
+            } ?: listOf("Analyze an image to see detection time"),
+            modifier = Modifier.padding(paddingValues)
+        )
+    }
+}
+
+private fun formatTime(time: Double) = "%5f".format(time)
+
+private const val MIN_RUN_TIMES = 1
+
+@Composable
+private fun RunConfigurationDialog(open: Boolean, onDismiss: () -> Unit, onConfirm: (Int) -> Unit) {
```

Comment:
The same question about `Boolean` naming.

Reply:
I'll rename it to `opened` by analogy with `expanded` in Compose's `ExposedDropdownMenu`, for example